agent/logger/log.go

```diff
@@ -53,24 +54,32 @@ var Config *logConfig
 
 func logfmtFormatter(params string) seelog.FormatterFunc {
 	return func(message string, level seelog.LogLevel, context seelog.LogContextInterface) interface{} {
-		return fmt.Sprintf(`level=%s time=%s msg=%q module=%s
-`, level.String(), context.CallTime().UTC().Format(time.RFC3339), message, context.FileName())
+		cc, ok := context.CustomContext().(map[string]string)
+		var customContext string
+		if ok && len(cc) > 0 {
+			var sortedContext []string
+			for k, v := range cc {
+				sortedContext = append(sortedContext, k+"="+v)
+			}
+			sort.Strings(sortedContext)
+			customContext = " " + strings.Join(sortedContext, " ")
+		}
+		return fmt.Sprintf(`level=%s time=%s msg=%q module=%s%s
+`, level.String(), context.CallTime().UTC().Format(time.RFC3339), message, context.FileName(), customContext)
```

Comment:
this and lines 74-82 share enough in common that I'd like to see them as a separate function.  (You might have done this in one of the later commits... If so, please disregard. I'm going through in chronological order.)